### ISSUE TITLE
coffee-script (with dash) project on npm is at 1.12.7

### ISSUE THREAD

**Issue by aral** [jashkenas/coffeescript#4711]
Not a biggie, but I thought you might want to update it to 2.0.0, @GeoffreyBooth :)

https://www.npmjs.com/package/coffee-script

**Comment by vendethiel** [comment]
The package should issue a deprecation warning, right?

**Comment by aral** [comment]
@vendethiel I just saw it in the npm search (I switched to using the coffeescript package). If it does have a deprecation warning, please feel free to close. (Do deprecation notices display on nom? I did just look and couldn’t see one for coffee-script – or I managed to miss it.)

**Comment by GeoffreyBooth** [comment]
I'm planning to push a 1.12.8 that includes an install message telling people to update their `package.json` files to use `coffeescript`. I don't think we want to maintain two modules.

**Comment by jashkenas** [comment]
Let's continue to be careful with how/when/where you want to push people towards moving things over to `coffeescript`.

Currently, `coffee-script` had "1,232,706 downloads in the last week", and `coffeescript` had 24,046, so there's still a ways to go yet.

**Comment by GeoffreyBooth** [comment]
Yeah I'm in no hurry. I want to see a few weeks go by without deadly bug reports against 2 before we start pushing people toward it.

**Comment by lydell** [comment]
Perhaps those who use CoffeeScript as bash don't want to be pushed toward 2 at all.

**Comment by GeoffreyBooth** [comment]
> Perhaps those who use CoffeeScript as bash don’t want to be pushed toward 2 at all.

Actually quite the opposite. Node 8 natively supports every feature of CoffeeScript 2 except for `import`/`export` statements (but you can just use CommonJS `require`/`exports` instead) and JSX (but I don’t see how that would be of much use for DevOps). And CoffeeScript 2 adds async functions, which are awfully handy for DevOps, and object destructuring, which is great for merging options objects together. The other nice thing about CS2 is that debugging should be easier, since the generated JavaScript should be so much closer to the original Coffee and therefore if Node prints snippets from thrown exceptions or syntax errors, the snippets will be more recognizable without necessarily needing to refer to line numbers.

Even if a shell-scripter were stuck in Node <=4 for some reason, they could just use CS2 with `--transpile`. As I said on another thread, I really don’t see any advantages of CS1 over CS2.

**Comment by bazineta** [comment]
I strongly recommend **not** forcing the 'coffee-script' version to be 2.x, but rather leaving it at the 1.x version, with a notice that 2.x is available.

Certainly, it's possible to move to 2.x, but the needs of maintaining a production application typically mean that people are unable to move instantly.

Yes, there are many nice features, we all greatly appreciate them, and we're looking forward to the migration. However, we can't accomplish that in zero time, and if things are simply arbitrarily broken, there will be a loss of mindshare as a result.

**Comment by GeoffreyBooth** [comment]
Actually there’s one thing I forgot: if you can’t run Node 6+, you _need_ to run CS1, as the CS2 compiler itself won’t run in Node <= 4. So I guess there’s that use case.

@bazineta it’s not like the 1.x versions go away. Someone can always `npm install coffeescript@1` or `npm install coffee-script@1`, and will be able to for all time (or for all the time that NPM is with us and preserves their current never-delete-versions policy, whichever comes first).

**Comment by bazineta** [comment]
@GeoffreyBooth There are products such as the Keymetrics pm2 utility that are using the two names to determine which version the user requires when installing transpilation support.

Additionally, as you mention, not everyone runs the bleeding-edge version of Node, and anyone running v6 is going to need additional, presumably babel, transpilation support, are they not?

Honestly, guys, you've gone to heroic efforts here to bring this forward, but if you screw over everyone who's running coffee in production by borking things that have made heretofore reasonable assumptions, it's like tripping at the finish line.

**Comment by GeoffreyBooth** [comment]
It’s not a reasonable assumption that just `npm install coffee-script` will always install a version of CoffeeScript that is compatible with your environment or project. You really should use `npm install coffeescript@1.12.7` or whatever specific version you know works, or update your `package.json` to `"coffeescript": "1.12.7"` or whatever version you know works.

**Comment by bazineta** [comment]
Of that there is no doubt or argument. However, the hard fact is that this is a breaking change, and there's a very large amount of tooling and third party stuff out there that's going to need some time to react. This is the first time anything dependent on CS has really needed to deal with breaking changes of this magnitude, and a lot of assumptions are about to be tested as being poor ones.

If you're just adamant about breaking things and driving people to updates that they may not be well positioned for at this time, well, I guess I've said my piece, and you've frankly made what was a hard decision easier for me.

**Comment by GeoffreyBooth** [comment]
Closing in favor of #4714.
